src/corvus.c

```diff
@@ -18,14 +18,33 @@
 #include "timer.h"
 #include "alloc.h"
 
+static pthread_spinlock_t signal_lock;
 static struct context *contexts;
 
-void sig_handler(int sig)
+void sigsegv_handler(int sig)
 {
-    int i;
+    if (pthread_spin_trylock(&signal_lock) == EBUSY) {
+        sigset_t mask;
+        sigemptyset(&mask);
+        pselect(0, NULL, NULL, NULL, NULL, &mask);
```

Comment:
First of all I do think the original implementation is safe enough:
```c
        sigset_t mask;
        sigemptyset(&mask);
        pselect(0, NULL, NULL, NULL, NULL, &mask);
```

There are many ways to make the 'EBUSY' thread more safer:
```c
        // block all signals exclude sigkill and sigstop
        sigset_t mask;
        sigfillset(&mask);
        pthread_sigmask(SIG_SETMASK, &mask, NULL);
        pselect(0, NULL, NULL, NULL, NULL, &mask);

        // or
        sigset_t mask;
        sigemptyset(&mask);
        while (true) {
            pselect(0, NULL, NULL, NULL, NULL, &mask);
        }

       // ...
```
In fact the handling of sigsegv is for production and not only for developing.